pkg/ccl/streamingccl/streamingest/external_connection.go

```diff
@@ -0,0 +1,55 @@
+// Copyright 2022 The Cockroach Authors.
+//
+// Licensed as a CockroachDB Enterprise file under the Cockroach Community
+// License (the "License"); you may not use this file except in compliance with
+// the License. You may obtain a copy of the License at
+//
+//     https://github.com/cockroachdb/cockroach/blob/master/licenses/CCL.txt
+
+package streamingest
+
+import (
+	"context"
+	"net/url"
+
+	"github.com/cockroachdb/cockroach/pkg/ccl/streamingccl/streamclient"
+	"github.com/cockroachdb/cockroach/pkg/cloud/externalconn"
+	"github.com/cockroachdb/cockroach/pkg/cloud/externalconn/connectionpb"
+	"github.com/cockroachdb/cockroach/pkg/security/username"
+)
+
+// The canonical PostgreSQL URL scheme is "postgresql", however our
+// own client commands also accept "postgres".
+func postgresSchemes() [2]string {
+	return [2]string{"postgres", "postgresql"}
+}
+
+func validatePostgresConnectionURI(
+	ctx context.Context, execCfg interface{}, user username.SQLUsername, uri string,
+) error {
+	parsedURI, err := url.Parse(uri)
+	if err != nil {
+		return err
+	}
+	conn, err := streamclient.NewPartitionedStreamClient(ctx, parsedURI)
+	if err != nil {
+		return err
+	}
+	if err = conn.Dial(ctx); err != nil {
+		return err
+	}
+	return conn.Close(ctx)
+}
+
+func init() {
+	for _, scheme := range postgresSchemes() {
+		externalconn.RegisterConnectionDetailsFromURIFactory(
+			scheme,
+			connectionpb.ConnectionProvider_sql,
```

Comment:
heh, so I think we added this for `experimental-sql` but didn't end up using it because of this TODO https://github.com/cockroachdb/cockroach/blob/2f811420a1492ca547659003f193233341b7fd87/pkg/ccl/changefeedccl/sink_external_connection.go#L112. I don't now much about experimental sql sinks but does it makes sense to reuse this type or should we delete it and add a `ConnectionProvider_postgres`. On a similar note, I don't think this ExternalConnection object should be of type `Storage`: https://github.com/cockroachdb/cockroach/blob/eb3e5605c3f3fcef801589c96af4f923b64d4078/pkg/cloud/externalconn/connectionpb/connection.go#L24. Should we add a new type?